### PR TITLE
Migrate to new IO library.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/dns-simple-server.toit [hunk 1]
```diff
@@ -48,45 +48,47 @@ class SimpleDnsServer:
...
+        q-type := reader-be.read-uint16
+        q-class := reader-be.read-uint16
         if q-class == dns.CLASS-INTERNET and q-type == dns.RECORD-A:
           response.resource-record q-name
         else:
           return response.create-error_ dns.ERROR-NOT-IMPLEMENTED
 
       // Reread the query packet.
-      position = 12
+      reader = io.Reader query
```
Comment (erikcorry) [inline src/dns-simple-server.toit:82]: Perhaps we could add a reset function.

File: src/dns-simple-server.toit [hunk 2]
```diff
@@ -2,8 +2,8 @@
 // Use of this source code is governed by an MIT-style license that can be
 // found in the LICENSE file.
 
-import binary show BIG-ENDIAN
-import bytes show Buffer
+import io show BIG-ENDIAN
```
Comment (erikcorry) [inline src/dns-simple-server.toit]: Do we still use this?

**Review by erikcorry — APPROVED** [review]

File: src/dns-simple-server.toit (hunk 2 above)
Comment (floitsch) [reply src/dns-simple-server.toit]: I had to update the `get` function which was still using it, but it's not used anymore.
done.

File: src/dns-simple-server.toit (hunk 1 above)
Comment (floitsch) [reply src/dns-simple-server.toit:82]: As discussed in person.
Not that easy. -> leaving as is.

**Pull request merged**